server/src/main/java/com/broadcom/lsp/cobol/service/CobolTextDocumentService.java

```diff
@@ -248,8 +262,13 @@ private void analyzeDocumentFirstTime(String uri, String text, boolean userReque
               ofNullable(docs.get(uri)).ifPresent(doc -> doc.setAnalysisResult(result));
               publishResult(uri, result, copybookProcessingMode);
               outlineMap.get(uri).complete(result.getOutlineTree());
-            })
-        .whenComplete(reportExceptionIfThrown(createDescriptiveErrorMessage("analysis", uri)));
+              completableFuture.complete(null);
```

Comment:
Isn't it better complete with some meaning meaning message, complete("Analyze-completed") ?    CompletableFuture<String>

Reply:
CompletableFuture<Void> , so returning null. Anyways we need it only for chaining. I will hold this PR as suggested for tomorrow's session.